lib/llm/src/migration.rs

```diff
@@ -503,7 +495,8 @@ mod tests {
         let next_generate: ServerStreamingEngine<PreprocessedRequest, Annotated<LLMEngineOutput>> =
             mock_engine;
 
-        let mut retry_manager = RetryManager::build(context_id, request, next_generate, 0)
+        let ctx = Arc::new(Controller::new(context_id.clone()));
```

Comment:
this code repeats several times, is there a better way to do this?

Reply:
I think it's fine as-is, but would be a good "pick issue" for somebody to fixup in a separate PR.